examples/trait-incrementer/lib.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright 2018-2021 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::new_without_default)]
+
+use ink_lang as ink;
+
+/// Allows to increment and get the current value.
+#[ink::trait_definition]
+pub trait Increment {
+    /// Increments the current value of the implementer by one (1).
```

Comment:
My nitpick here was that it's up to the trait implementer to choose the amount `inc` increases by 😉

Reply:
I do not agree because this would violate the invariant stated by the docs of the trait's message.

Reply:
Just because our concrete incrementer does _also_support `inc_by` does not mean this invariant must not be uphold for the trait implementation in particular.